### DISCUSSION TITLE
Support workspace / buildtool related source change detection

### DISCUSSION THREAD

**Discussion by lefou** [build-server-protocol/build-server-protocol#502] [Proposals]
As an implementer of BSP, Mill has no way to tell the BSP client, which source files would affect the workspace. E.g editing `build.sc` or some included file. If those files are edited and saved, a `workspace/reload` should be initiated, but a BSP client is currently either forced to guess if it should reload the workspace or, even worse, depends on the user to manually reload the project.

To underpin my statement, in Metals, there is some heuristics applied (`scala.meta.internal.metals.MetalsLanguageServer#onBuildChangedUnbatched` and `scala.meta.internal.builds.BuildTools`) , to decide whether a workspace reload is needed.

**Comment by jastice** [comment]
Agree we should have this.

If the build tool is watching its own sources, however, it can send a [`buildTarget/didChange`](https://build-server-protocol.github.io/docs/specification.html#build-target-changed-notification) notification to tell the client to reload.

**Comment by ckipp01** [comment]
This sounds like a great idea. One thing that does come to mind though is that how would this work for a build server that isn't the same tool as the build tool. I can see this working smoothly for sbt and Mill, but what about Bloop? Since even the files that Bloop is watching isn't necessarily your build files, for example I don't think Bloop would know anything about your `build.sbt` file right?

**Comment by olafurpg** [comment]
@ckipp01 it would be the responsibility of Bloop to register the appropriate file patterns for sbt/Mill.
